modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbTableStorage.java

```diff
@@ -238,65 +254,94 @@ public void destroy() throws StorageException {
     /** {@inheritDoc} */
     @Override
     public PartitionStorage getOrCreatePartition(int partId) throws StorageException {
-        PartitionStorage partition = getPartition(partId);
+        assert !stopped : "Storage has been stopped";
 
-        if (partition != null) {
-            return partition;
-        }
+        synchronized (partitionsLock) {
```

Comment:
Will this prevent us from concurrent partitions creation? I guess so, this is not good. Ignite 2.x uses `org.apache.ignite.internal.util.GridStripedLock` for similar purposes, you could port it :)
Although I'm not sure that `Runtime.getRuntime().availableProcessors()` is good choice for stripes count